contracts/PosterFactory.sol

```diff
@@ -0,0 +1,96 @@
+//SPDX-License-Identifier: Unlicense
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/utils/CountersUpgradeable.sol";
+import "./metadata/ERC1155OnChainMetadata.sol";
+
+contract PosterFactory is ERC1155OnChainMetadata {
+    using CountersUpgradeable for CountersUpgradeable.Counter;
+
+    // Index of current PosterID.
+    CountersUpgradeable.Counter private tokenId;
+
+    function initialize(
+        string memory name_,
+        string memory symbol_,
+        string memory contract_description_,
+        string memory contract_image_,
+        string memory contract_external_link_,
+        uint256 contract_seller_fee_basis_points_,
+        address royaltyRecipient_
+    ) external initializer {
+        __ERC1155_init("");
+        __Context_init();
+        __ERC165_init();
+        name = name_;
+        symbol = symbol_;
+        contract_description = contract_description_;
+        contract_image = contract_image_;
+        contract_external_link = contract_external_link_;
+        contract_seller_fee_basis_points = contract_seller_fee_basis_points_;
+        contract_fee_recipient = royaltyRecipient_;
+    }
+
+    /**
+     * @dev checks caller can create a new poster.
+     * @param _maxSupply amount to supply the first owner
+     */
+    modifier createPreCheck(uint256 _maxSupply) {
+        uint256 price = 5000000000000 * _maxSupply;
```

Comment:
price is hardcoded? Let's let the creator set the price.

Reply:
price is now set in the `initializer` ✅ 
- passed in `tasks/deploy.js`

<img width="479" alt="Screen Shot 2022-07-07 at 1 27 44 PM" src="https://user-images.githubusercontent.com/23249402/177823895-33873141-f5f3-42c7-a54a-745e2731fe62.png">